concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -195,25 +195,25 @@ emptyBlockTable = BlockTable emptyDeadCache HM.empty
 -- The first type parameter, @pv@, is the protocol version.
 -- The second type parameter, @ati@, is a type determining the account transaction index to use.
 -- The third type parameter, @bs@, is the type of block states.
-data SkovPersistentData (pv :: ProtocolVersion) (bs :: Type) = SkovPersistentData
+data SkovPersistentData (pv :: ProtocolVersion) = SkovPersistentData
     { -- |Map of all received blocks by hash.
-      _blockTable :: !(BlockTable pv bs),
+      _blockTable :: !(BlockTable pv (PBS.HashedPersistentBlockState pv)),
       -- |Map of (possibly) pending blocks by hash
       _possiblyPendingTable :: !(HM.HashMap BlockHash [PendingBlock]),
       -- |Priority queue of pairs of (block, parent) hashes where the block is (possibly) pending its parent, by block slot
       _possiblyPendingQueue :: !(MPQ.MinPQueue Slot (BlockHash, BlockHash)),
       -- |Pointer to the last finalized block
-      _lastFinalized :: !(PersistentBlockPointer pv bs),
+      _lastFinalized :: !(PersistentBlockPointer pv (PBS.HashedPersistentBlockState pv)),
```

Comment:
Is there any reason why BlockPointer/PersistentBlockPointer still has the `s` parameter?

Reply:
Right, so I tried removing it, but it started to complain about cyclic module dependencies, so I left it there.
I could look more into removing it, but then I would make it a separate PR.

Reply:
Ok. Let's leave it.